src/main/java/core/basesyntax/model/User.java

```diff
@@ -3,11 +3,19 @@
 import java.util.Objects;
 
 public class User {
+    private static Long idCounter = 1L;
```

Comment:
The `idCounter` is a static field that increments with each new `User` instance. This approach works but may lead to issues in a multi-threaded environment or if the application restarts. Consider using a more robust ID generation strategy, such as UUIDs or a database sequence, if this is a concern.